lib/streams/compound-duplex.js

```diff
@@ -63,12 +63,40 @@ function CompoundDuplex(writable, readable, options) {
 		self.push(null);
 	});
 
-	this.on('finish', function() {
-		writable.end();
+	readable.on('finish', function() {
```

Comment:
I see now that you're emitting the "EOF" in _duplexPrefinish.  But still, you probably shouldn't be listening to 'finish' on a readable stream (when readables don't even emit finish) and you probably shouldn't be manually emitting 'finish' (it's an event emitted from the streams internals, and I've had problems in the past when I've tried to emit it manually ... internal stream state seems to get desync'd or something).